src/main/java/frc/robot/subsystems/swerve/odometryThread/PhoenixOdometryThread6328.java

```diff
@@ -0,0 +1,192 @@
+ * This also allows Phoenix Pro users to benefit from lower latency between devices using CANivore
+ * time synchronization.
+ */
+public class PhoenixOdometryThread6328 extends Thread {
+
+    private final Lock SIGNALS_LOCK = new ReentrantLock();
+    private final List<Queue<Double>> queues = new ArrayList<>();
+    private final Queue<Double> timestamps = new ArrayBlockingQueue<>(OdometryThreadConstants.MAX_QUEUE_SIZE);
+    private StatusSignal[] signals = new StatusSignal[0];
+    private boolean[] isLatencySignals = new boolean[0];
+    private boolean isCANFD = false; //todo check why it is not per signal
+
+    private static PhoenixOdometryThread6328 INSTANCE = null;
+
+    public static PhoenixOdometryThread6328 getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new PhoenixOdometryThread6328();
+        }
+        return INSTANCE;
+    }
+
+    private PhoenixOdometryThread6328() {
+        setName("PhoenixOdometryThread");
+        setDaemon(true);
+        start();
+    }
+
+
+    public Queue<Double> getTimestampQueue() {
+        return timestamps;
+    }
+
+
+    public Queue<Double> registerLatencySignal(ParentDevice device, StatusSignal<Double> signal, StatusSignal<Double> signalSlope) {
+        return registerSignals(true, device, new StatusSignal[]{signal, signalSlope});
+    }
+
+    public Queue<Double> registerRegularSignal(ParentDevice device, StatusSignal<Double> signal) {
+        return registerSignals(false, device, new StatusSignal[]{signal});
+    }
+
+    private Queue<Double> registerSignals(boolean isLatencySignal, ParentDevice device, StatusSignal<Double>[] signals) {
+        Queue<Double> queue = new ArrayBlockingQueue<>(OdometryThreadConstants.MAX_QUEUE_SIZE);
+        SIGNALS_LOCK.lock();
+        SWERVE.ODOMETRY_LOCK.lock();
+        try {
+            isCANFD = CANBusJNI.JNI_IsNetworkFD(device.getNetwork());
+            for (StatusSignal<Double> signal : signals) {
+                registerSignal(signal);
+            }
+            updateIsLatencySignals(isLatencySignal);
+            queues.add(queue);
+        }
+        finally {
+            SIGNALS_LOCK.unlock();
+            SWERVE.ODOMETRY_LOCK.unlock();
+        }
+        return queue;
+    }
+
+    private void registerSignal(StatusSignal<Double> signal) {
+        StatusSignal[] newSignals = new StatusSignal[signals.length + 1];
+        System.arraycopy(signals, 0, newSignals, 0, signals.length);
+        newSignals[signals.length] = signal;
+        signals = newSignals;
+    }
+
+    private void updateIsLatencySignals(boolean isLatencySignal) {
+        boolean[] newIsLatencySignals = new boolean[isLatencySignals.length + 1];
+        System.arraycopy(isLatencySignals, 0, newIsLatencySignals, 0, isLatencySignals.length);
+        newIsLatencySignals[isLatencySignals.length] = isLatencySignal;
+        isLatencySignals = newIsLatencySignals;
```

Comment:
same here

Reply:
it is better for memory and runtime and stuff, it's not a problem to use array here

Reply:
im sure you could find/make an object that does not use much memory or runtime, something like a Yeda Node.